### PR TITLE
marketplace: change redhatsubscriptions model to return a list of customer ids (PROJQUAY-7129)

### PR DESCRIPTION
**Pull request by Marcusk19**
Changes model for redhatsubscriptions table to select all customer ids for a quay user and return them as an array.

Takes into account that a quay user can have multiple different web customer ids.

### CODE REVIEW

**Comment by Marcusk19** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/quay/quay/2868?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2868** <a href="https://app.graphite.dev/github/pr/quay/quay/2868?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2867** <a href="https://app.graphite.dev/github/pr/quay/quay/2867?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Marcusk19 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: data/model/entitlements.py [hunk 1]
```diff
@@ -6,10 +6,12 @@
...
 
-def get_web_customer_id(user_id):
+def get_web_customer_ids(user_id):
     try:
-        query = RedHatSubscriptions.get(RedHatSubscriptions.user_id == user_id).account_number
-        return query
+        customer_ids = []
+        for customer in RedHatSubscriptions.select().where(RedHatSubscriptions.user_id == user_id):
+            customer_ids.append(customer.account_number)
+        return customer_ids
```
Comment (kleesc) [inline data/model/entitlements.py:14]: You can replace this with a list comprehension:
```
[
  customer.account_number 
  for customer in RedHatSubscriptions.select().where(RedHatSubscriptions.user_id == user_id)
]`

File: data/model/entitlements.py [hunk 2]
```diff
@@ -6,10 +6,12 @@
...
     except RedHatSubscriptions.DoesNotExist:
         return None
```
Comment (kleesc) [inline data/model/entitlements.py:16]: Do you still want to return a null value or an empty list with this change?

File: util/marketplace.py [hunk 3]
```diff
@@ -22,15 +22,14 @@ def __init__(self, app_config):
 
     def get_account_number(self, user):
         email = user.email
-        account_number = entitlements.get_web_customer_id(user.id)
-        if account_number is None:
+        account_numbers = entitlements.get_web_customer_ids(user.id)
+        if len(account_numbers) == 0:
```
Comment (kleesc) [inline util/marketplace.py:26]: See comment above. This won't work if a null value is returned.

Comment (Marcusk19) [reply util/marketplace.py:26]: Ok, probably better to expect to return a null value. Other parts of the code may be expecting a null value as well anyways

**Review by kleesc — APPROVED** [review]
LGTM

**Pull request merged**